### PR TITLE
[Modules] Updated Network/Connections to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Network/Connections to new dependency approach

| Pipeline |
| - |
| [Network: Connections](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.connections.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Network/connections/.test/vnet2vnet/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,64 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/connections/.test/vnet2vnet/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,132 @@
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the primary Public IP to create.')
+param primaryPublicIPName string
+
+@description('Required. The name of the primary VNET to create.')
+param primaryVirtualNetworkName string
+
+@description('Required. The name of the primary VNET Gateways to create.')
```
Comment (eriqua) [inline modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep:10]:
```suggestion
@description('Required. The name of the primary VNET Gateway to create.')
```

File: modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+param primaryVirtualNetworkGateway string
+
+@description('Required. The name of the secondary Public IP to create.')
+param secondaryPublicIPName string
+
+@description('Required. The name of the secondary VNET to create.')
+param secondaryVirtualNetworkName string
+
+@description('Required. The name of the secondary VNET Gateways to create.')
```
Comment (eriqua) [inline modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep:19]:
```suggestion
@description('Required. The name of the secondary VNET Gateway to create.')
```

File: modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        vpnType: 'RouteBased'
+        vpnGatewayGeneration: 'Generation2'
+        sku: {
+            name: 'VpnGw2'
+            tier: 'VpnGw2'
+        }
+    }
+}
+
+@description('The resource ID of the first created Virtual Network Gateways.')
```
Comment (eriqua) [inline modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep:128]:
```suggestion
@description('The resource ID of the primary Virtual Network Gateway.')
```

File: modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+output primaryVNETGatewayResourceID string = primaryVNETGateway.id
+
+@description('The resource ID of the second created Virtual Network Gateways.')
```
Comment (eriqua) [inline modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep:131]:
```suggestion
@description('The resource ID of the secondary Virtual Network Gateway.')
```

File: modules/Microsoft.Network/connections/.test/vnet2vnet/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  location: location
+}
+
+module resourceGroupResources 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-paramNested'
+  params: {
+    primaryPublicIPName: 'dep-<<namePrefix>>-pip-${serviceShort}-1'
+    primaryVirtualNetworkName: 'dep-<<namePrefix>>-vnet-${serviceShort}-1'
+    primaryVirtualNetworkGateway: 'dep-<<namePrefix>>-vpn-gw-${serviceShort}-1'
```
Comment (eriqua) [inline modules/Microsoft.Network/connections/.test/vnet2vnet/deploy.test.bicep:37]: Should this be
```suggestion
    primaryVirtualNetworkGatewayName: 'dep-<<namePrefix>>-vpn-gw-${serviceShort}-1'
```

File: modules/Microsoft.Network/connections/.test/vnet2vnet/deploy.test.bicep [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    secondaryPublicIPName: 'dep-<<namePrefix>>-pip-${serviceShort}-2'
+    secondaryVirtualNetworkName: 'dep-<<namePrefix>>-vnet-${serviceShort}-2'
+    secondaryVirtualNetworkGateway: 'dep-<<namePrefix>>-vpn-gw-${serviceShort}-2'
```
Comment (eriqua) [inline modules/Microsoft.Network/connections/.test/vnet2vnet/deploy.test.bicep:40]:
```suggestion
    secondaryVirtualNetworkGatewayName: 'dep-<<namePrefix>>-vpn-gw-${serviceShort}-2'
```

File: modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+param primaryVirtualNetworkGateway string
+
+@description('Required. The name of the secondary Public IP to create.')
+param secondaryPublicIPName string
+
+@description('Required. The name of the secondary VNET to create.')
+param secondaryVirtualNetworkName string
+
+@description('Required. The name of the secondary VNET Gateways to create.')
+param secondaryVirtualNetworkGateway string
```
Comment (eriqua) [inline modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep:20]:
```suggestion
param secondaryVirtualNetworkGatewayName string
```

File: modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+param location string = resourceGroup().location
+
+@description('Required. The name of the primary Public IP to create.')
+param primaryPublicIPName string
+
+@description('Required. The name of the primary VNET to create.')
+param primaryVirtualNetworkName string
+
+@description('Required. The name of the primary VNET Gateways to create.')
+param primaryVirtualNetworkGateway string
```
Comment (eriqua) [inline modules/Microsoft.Network/connections/.test/vnet2vnet/dependencies.bicep:11]:
```suggestion
param primaryVirtualNetworkGatewayName string
```

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**